src/Caller.sol

```diff
@@ -215,6 +222,18 @@ contract Caller is EIP712("Caller", "1"), ERC2771Context(address(this)) {
         return _call(sender, target, data, msg.value);
     }
 
+    /// @notice Sets the new nonce for the sender.
+    /// @param newNonce The new nonce.
+    /// It must be larger than the current nonce but by no more than MAX_NONCE_INCREASE.
+    function setNonce(uint256 newNonce) public {
+        address sender = _msgSender();
```

Comment:
Currently, with the `approve` pattern an authorized address would be able to call `setNonce` as well and basically prevent the actual owner from using the Caller.

Again, we already had all these trust discussions but I wanted to point out this fact. 

I think you anyway also added a test case for it.

Reply:
That's a good observation, I've been thinking about it for a while and I think that `setNonce` callable by the operators is the right design. It makes `setNonce` batchable and scriptable just like any other operations, which is nice. In case of an emergency (an authorized user going rogue) the original user can always call Caller directly, without a signature which could be blocked, after all only EOAs can generate signed calls. I'm not sure if in the real world this scenario will be common, I can't think of a good reason to have free will (human) operators and using signed calls for delegating execution to anonymous 3rd parties.